crates/components-runtime/src/context.rs

```diff
@@ -32,41 +36,85 @@ impl ComponentsContext {
         let mut linker = Linker::new(&engine);
         wasmtime_wasi::add_to_linker_async(&mut linker)?;
 
-        let components = config
+        // Data collection components
+        let data_collection_components = config
             .data_collection
             .iter()
             .map(|entry| {
                 let span = tracing::info_span!("component-context", component = %entry.name);
```

Comment:
Maybe add the component category here as a span attribute instead of in the trace messages?

```suggestion
                let span = tracing::info_span!("component-context", component = %entry.name, category = "data-collection");
```